File: src/presets/minimal.ts
import { StatusLine, Git, Branch, Context, Tokens, Literal, Size } from '../dsl/index.js';
import type { SegmentNode } from '../types.js';

export const minimalPreset: SegmentNode[] = StatusLine(() => [
  { type: 'pwd.name', provider: 'pwd', style: { color: '39' } },
  Git({ prefix: ' | ', color: '240' })(() => [
    Branch({ color: 'whiteBright', bold: true }),
  ]),
  Context({ prefix: ' | ', color: '240' })(() => [
    Tokens({ color: 'white' }),
    Literal({ text: '/' }),
    Size({ color: 'white' }),
  ]),
]);

File: internal/preset/minimal.go
package preset

import "github.com/jheddings/ccnow/internal/types"

func minimalPreset() []types.SegmentNode {
	return []types.SegmentNode{
		{Type: "pwd.name", Provider: "pwd", Style: &types.StyleAttrs{Color: "39"}},
		{Type: "git.branch", Provider: "git", Style: &types.StyleAttrs{Color: "whiteBright", Bold: true, Prefix: " | "}},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " | "},
			Children: []types.SegmentNode{
				{Type: "context.tokens", Provider: "context", Style: &types.StyleAttrs{Color: "white"}},
				{Type: "context.size", Provider: "context", Style: &types.StyleAttrs{Color: "white", Prefix: "/"}},
			},
		},
	}
}
